### PR TITLE
added generatorTemplate to swagger docs in put patch and post

### PR DESCRIPTION
**Pull request by harshkothari410**

### CODE REVIEW

**Comment by harshkothari410** [comment]
Do not review it I found out one more thing is missing. I need to ask about that @iamigo

File: tests/api/v1/generators/post.js [hunk 1]
```diff
@@ -89,6 +89,21 @@ describe('tests/api/v1/generators/post.js >', () => {
     });
   });
 
+  it('post without Generator Teamplate error', (done) => {
```
Comment (iamigo) [inline tests/api/v1/generators/post.js]: typo in "Template"

**Comment by harshkothari410** [comment]
Yeah locally this test is not failing. Let me check.

File: api/v1/swagger.yaml [hunk 2]
```diff
@@ -3614,9 +3635,20 @@ paths:
...
+              generatorTemplate:
+                type: object
+                description: Details of Generator Template.
+                properties:
+                  name:
+                    type: string
+                    description: The Generator Template's name.
+                  version:
+                    type: string
+                    description: The Generator Template's version.
```
Comment (iamigo) [inline api/v1/swagger.yaml]: description should be:

`A Generator Template version number (MAJOR.MINOR.PATCH) or a valid x-range, tilde range or caret range. Any of X, x, or * may be used to "stand in" for one of the numeric major, minor or patch values. A tilde range allows patch-level changes if a minor version is specified on the comparator, or allows minor-level changes if not. A caret range allows changes that do not modify the left-most non-zero major, minor or patch number. See https://github.com/npm/node-semver for more details on semver matching.`

File: api/v1/swagger.yaml [hunk 3]
```diff
@@ -2040,7 +2040,6 @@ paths:
           in: formData
           description: The unique readable name for the bot
           type: string
-          required: true
```
Comment (iamigo) [inline api/v1/swagger.yaml]: careful! you don't want to change the bot stuff in this pr!

Comment (harshkothari410) [reply api/v1/swagger.yaml]: Oops I haven't change anything. Maybe because I pulled and pushed. This is strange.

**Comment by harshkothari410** [comment]
@iamigo fixed that.

**Comment by harshkothari410** [comment]
But we are not validating version at swagger level. So writing tests about version validation should not be part of this PR, and not even part of generators. It should be part of generatorTemplate. I can create another PR for these tests as part of GenetatorTemplate. Whats you thought? @iamigo

**Comment by iamigo** [comment]
ok we're doing that validation in the db layer

**Comment by harshkothari410** [comment]
Cool. So any other changes ?

**Comment by iamigo** [comment]
not from me... one more reviewer needed

**Pull request merged**